### PR TITLE
Fix tests for play_for verb

### PR DESCRIPTION
**Pull request by agalbachicar**
Goes on top of #1 

Should start reviewing from b7f2ae51b74a300a347008771243b885d72a972f onwards

This PR fixes and adds missing tests to the play_for verb (`Play::play(duration)`)

### CODE REVIEW

**Comment by agalbachicar** [comment]
@gbiggs for review.

File: rosbag2_transport/test/rosbag2_transport/test_play_for.cpp [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+    player_ = std::make_shared<rosbag2_transport::Player>(
+      std::move(
+        reader), storage_options_, play_options_);
+  }
+
+  std::shared_ptr<rosbag2_transport::Player> player_;
+  std::future<void> await_received_messages_;
+};
+
+TEST_F(RosBag2PlayForTestFixture, play_for_recorded_messages_are_played_for_all_topics)
```
Comment (gbiggs) [inline rosbag2_transport/test/rosbag2_transport/test_play_for.cpp]:
```suggestion
TEST_F(RosBag2PlayForTestFixture, play_for_all_are_played_due_to_duration)
```

**Review by gbiggs — APPROVED** [review]
LGTM

**Comment by gbiggs** [comment]
OK to merge?

**Comment by agalbachicar** [comment]
> OK to merge?

Feel free to do so. I don't have the bits to do so (no write access). All the PRs are shown as approved but there is not merge button available. If you could do so, feel free to proceed with all.

**Comment by gbiggs** [comment]
Closes #1 #2 #3

**Comment by agalbachicar** [comment]
@gbiggs note that #2 was not included in this set of commits because it only depended on #1.

**Pull request merged**